cdap-proto/src/main/java/io/cdap/cdap/proto/security/Authorizable.java

```diff
@@ -289,7 +289,16 @@ private static void checkParts(EntityType entityType, List<String> parts, int in
           entityParts.put(entityType, parts.get(index - 1) + "." + parts.get(index));
         }
         break;
-      default:
+      case SYSTEM_APP_ENTITY:
+        if (parts.size() != 4 && index == (parts.size() - 1)) {
+          throw new IllegalArgumentException("Entity value is missing some parts or containing more parts. " +
+                                               "Expected: <namespace-name>.<app-name>.<entity-type>.<entity-name>, " +
```

Comment:
let's not use <entity-type> here, this will confuse user with our enum, better just use <type> or <app-entity-type>

Reply:
ack